account_payment_order/models/account_move_line.py

```diff
@@ -53,16 +53,7 @@ def _get_linked_move_communication(self):
         # Build a recordset to gather moves from which references have already
         # taken in order to avoid duplicates
         reference_moves = self.env["account.move"].browse()
-        # If we have credit note(s) - reversal_move_id is a one2many
-        if self.move_id.reversal_move_id:
-            references.extend(
```

Comment:
`references` doesn't contain all the needed ones. You are not calling `_get_payment_order_communication()` for this case.

Reply:
This condition is included in `_get_payment_order_communication()` method of `accont.move` (https://github.com/OCA/bank-payment/pull/1142#discussion_r1333933409).

`_get_communication()` method of `account.move.line` calls `_get_payment_order_communication()` so it already has this information.

Is there anything else I am not taking into account?

Reply:
But you are returning here other references directly hard-coded in https://github.com/OCA/bank-payment/pull/1153/files#diff-c44811bc7189570a631438d2dca538756d8ad24b26ced88bfd5df3ab44b74f60R67

Why this is not constant?

Reply:
Ok, it's already changed to be consistent.

Do you think the current behavior is correct?